numpy/core/tests/test_multiarray.py

```diff
@@ -92,6 +92,44 @@ def test_writeable(self):
         self.a[0] = 5
         self.a[0] = 0
 
+        # gh-9440 - make sure fromstring, from buffer on readonly buffers
+        # set writeable False
+        data = b'\x00' * 100
+        vals = np.frombuffer(data, 'B')
+        assert_raises(ValueError, vals.setflags, write=True)
+        types = np.dtype( [('vals', 'u1'), ('res3', 'S4')] )
+        values = np.core.records.fromstring(data, types)
+        vals = values['vals']
+        assert_raises(ValueError, vals.setflags, write=True)
+
+        data = bytearray(b'\x00' * 100)
```

Comment:
And perhaps add `def test_writeable_from_buffer` as a test name here

Reply:
done